e2e/stateful-pallet-storage/handlePaginated.test.ts

```diff
@@ -30,34 +31,49 @@ describe('📗 Stateful Pallet Storage Paginated', function () {
   let badMsaId: u64;
 
   before(async function () {
-    // Create a provider for the MSA, the provider will be used to grant delegation
-    [providerKeys, providerId] = await createProviderKeysAndId(fundingSource, 2n * DOLLARS);
+    [
+      // Create a provider for the MSA, the provider will be used to grant delegation
+      [providerKeys, providerId],
+      // Delegator Keys
+      delegatorKeys,
+      // Create a schema for Paginated PayloadLocation
+      schemaId,
+      // Create non supported schema
+      schemaId_unsupported,
+    ] = await Promise.all([
+      createProviderKeysAndId(fundingSource, 2n * DOLLARS),
+      createAndFundKeypair(fundingSource, 2n * DOLLARS),
+      getOrCreateAvroChatMessagePaginatedSchema(fundingSource),
+      ExtrinsicHelper.getOrCreateSchemaV3(
+        fundingSource,
+        AVRO_CHAT_MESSAGE,
+        'AvroBinary',
+        'OnChain',
+        [],
+        'test.handlePaginatedUnsupported'
+      ),
+    ]);
+
     assert.notEqual(providerId, undefined, 'setup should populate providerId');
     assert.notEqual(providerKeys, undefined, 'setup should populate providerKeys');
 
-    // Create a schema for Paginated PayloadLocation
-    schemaId = await getOrCreateAvroChatMessagePaginatedSchema(providerKeys);
-
-    // Create non supported schema
-    schemaId_unsupported = await ExtrinsicHelper.getOrCreateSchemaV3(
-      providerKeys,
-      AVRO_CHAT_MESSAGE,
-      'AvroBinary',
-      'OnChain',
-      [],
-      'test.handlePaginatedUnsupported'
-    );
-
-    // Create a MSA for the delegator and delegate to the provider
-    [delegatorKeys, msa_id] = await createDelegatorAndDelegation(fundingSource, schemaId, providerId, providerKeys);
-    assert.notEqual(msa_id, undefined, 'setup should populate msa_id');
+    [
+      // Create a MSA for the delegator and delegate to the provider
+      [delegatorKeys, msa_id],
+      // Create an MSA that is not a provider to be used for testing failure cases
+      [badMsaId],
+    ] = await Promise.all([
+      // createDelegatorAndDelegation IS FAILING DUE TO FUNDING!!!! (PRE-delegatorKeys at least)
```

Comment:
Is this still true? Should this still be here/ does it need more work?

Reply:
No it is not. Fixing...
```suggestion
```